src/NationalInstruments.Analyzers/Style/ChainOfMethodsWithLambdasAnalyzer.cs

```diff
@@ -0,0 +1,110 @@
+using System.Collections.Immutable;
+using System.Linq;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.CSharp;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using Microsoft.CodeAnalysis.Diagnostics;
+using NationalInstruments.Analyzers.Properties;
+using NationalInstruments.Analyzers.Utilities;
+using NationalInstruments.Analyzers.Utilities.Extensions;
+
+namespace NationalInstruments.Analyzers.Style
+{
+    /// <summary>
+    /// Enforces the convention -
+    /// https://github.com/ni/csharp/blob/main/docs/Coding-Conventions.md#f34-%EF%B8%8F-consider-splitting-long-chains-of-dotted-methodproperty-invocations
```

Comment:
@jryckman are we going to update this rule in the convention page to mention specifically about lambdas? Or is that going to be a different rule? 

If it's going to be a different rule then I will remove the reference from here.

Reply:
we should update the documentation in the convention page to specifically call out lambdas, but also add a note about readability as well

Reply:
Sure, I'll update the convention page in a follow up PR.